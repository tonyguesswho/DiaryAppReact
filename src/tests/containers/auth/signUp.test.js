import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import SignUpComponent, { SignUp } from '../../../containers/auth/SignUp';
import Root from '../../../root';

let wrapped;
let wrapper;

const props = {
  clearError: jest.fn(),
  signup: jest.fn()
}

beforeEach(() => {
  wrapped = mount(
    <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
      <Root>
        <SignUpComponent
          clearError={props.clearError}
          signup={props.signup}
        />
      </Root>
    </MemoryRouter>
  );

  wrapper = shallow(
    <SignUp
      clearError={props.clearError}
      signup={props.signup}
    />
  )
});

afterEach(() => wrapped.unmount());

describe('Signup UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });

    it('has 5 input fields and a button', () => {
      expect(wrapped.find('input').length).toEqual(5);
    });

    it('should simulate signup process', () => {
      const Signin = wrapper.find('form');

      const mockedEvent = {
        preventDefault: jest.fn()
      }

      Signin.simulate('submit', mockedEvent)
    });

    it('should simulate onChange validation process', () => {
      const Signin = wrapper.find('#password');

      const mockedEvent = {
        preventDefault: jest.fn(),
        target: {
          name: 'name',
          value: 'value'
        }
      }

      Signin.simulate('change', mockedEvent)
    });
  });
});
