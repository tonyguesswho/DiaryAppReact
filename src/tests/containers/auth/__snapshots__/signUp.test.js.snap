// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup UI render features container should render as expected 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <_default>
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Connect(withRouter(SignUp))
          clearError={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          signup={[MockFunction]}
        >
          <withRouter(SignUp)
            clearError={[Function]}
            errorMessage=""
            signup={[Function]}
          >
            <Route>
              <SignUp
                clearError={[Function]}
                errorMessage=""
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "testKey",
                        "pathname": "/",
                        "search": "",
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "testKey",
                      "pathname": "/",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/",
                    "search": "",
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                signup={[Function]}
              >
                <div
                  className="bg body"
                >
                  <header
                    className="header2"
                  >
                    <nav>
                      <ul
                        className="nav nav2"
                      >
                        <li>
                          <Link
                            replace={false}
                            to="/"
                          >
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              Home
                            </a>
                          </Link>
                        </li>
                        <li>
                          <Link
                            replace={false}
                            to="/signin"
                          >
                            <a
                              href="/signin"
                              onClick={[Function]}
                            >
                              Sign In
                            </a>
                          </Link>
                        </li>
                      </ul>
                    </nav>
                  </header>
                  <section
                    className="signup"
                    id="messageBox"
                  >
                    <div
                      className="box span31 span3-center card"
                    >
                      <h2>
                        Sign Up 
                      </h2>
                      <form
                        id="signUp"
                        onSubmit={[Function]}
                      >
                        <div
                          className="fullspan"
                        >
                          <label
                            htmlFor="email"
                          >
                            Email:
                          </label>
                        </div>
                        <div
                          className="fullspan"
                        >
                          <input
                            id="email"
                            name="email"
                            onChange={[Function]}
                            placeholder="email"
                            required={true}
                            type="email"
                          />
                        </div>
                        <div
                          className="fullspan"
                        >
                          <label
                            htmlFor="username"
                          >
                            username:
                          </label>
                        </div>
                        <div
                          className="fullspan"
                        >
                          <input
                            id="username"
                            name="username"
                            onChange={[Function]}
                            placeholder="username"
                            required={true}
                            type="text"
                          />
                        </div>
                        <div
                          className="fullspan"
                        >
                          <label
                            htmlFor="password"
                          >
                            Password:
                          </label>
                        </div>
                        <div
                          className="fullspan"
                        >
                          <input
                            id="password"
                            name="password"
                            onChange={[Function]}
                            placeholder="password"
                            required={true}
                            type="password"
                          />
                        </div>
                        <div
                          className="fullspan"
                        >
                          <label
                            htmlFor="cPassword"
                          >
                            Confirm Password:
                          </label>
                        </div>
                        <div
                          className="fullspan"
                        >
                          <input
                            id="confirmPassword"
                            name="confirmPassword"
                            onChange={[Function]}
                            placeholder="Confirm password"
                            required={true}
                            type="password"
                          />
                        </div>
                        <p
                          className="signin-prompt"
                        >
                          Already have an account
                          <Link
                            replace={false}
                            to="/signin"
                          >
                            <a
                              href="/signin"
                              onClick={[Function]}
                            >
                              Sign In
                            </a>
                          </Link>
                        </p>
                        <input
                          type="submit"
                          value="Sign Up"
                        />
                      </form>
                    </div>
                  </section>
                </div>
              </SignUp>
            </Route>
          </withRouter(SignUp)>
        </Connect(withRouter(SignUp))>
      </Provider>
    </_default>
  </Router>
</MemoryRouter>
`;
